### PR TITLE
Add Status Bar Api

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Adds Status Bar api through `ITopLevelWithPlatformStatusBar`, with color and visibility properties. The Api can be accessed on any Control instance, but is only applied when using the TopLevel's Content, ie the MainView. This Api is targeted toward mobile platforms.

New Properties in `Control`;
```csharp
StatusBarTheme? StatusBarTheme { get; set; }

bool? IsStatusBarVisible { get; set; }
```

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025421-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by Gillibald** [comment]
If this only applies to the TopLevel why are we defining the property on Control level?

File: src/Avalonia.Controls/TopLevel.cs [hunk 1]
```diff
@@ -75,6 +76,22 @@ public abstract class TopLevel : ContentControl,
         public static readonly StyledProperty<IBrush> TransparencyBackgroundFallbackProperty =
             AvaloniaProperty.Register<TopLevel, IBrush>(nameof(TransparencyBackgroundFallback), Brushes.White);
 
+        /// <summary>
+        /// Defines the StatusBarColor attached property.
+        /// </summary>
+        public static readonly AttachedProperty<Color?> StatusBarColorProperty =
```
Comment (danwalmsley) [inline src/Avalonia.Controls/TopLevel.cs:82]: we can put this in its own file:

`<UserControl MobileStatusBar.Color="Red" />`

**Review by danwalmsley — CHANGES_REQUESTED** [review]
we can refactor it a little so that most of the code doesnt go inside toplevel.cs (but still associated with it)

File: src/Avalonia.Controls/TopLevel.cs [hunk 2]
```diff
@@ -481,6 +482,8 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             _transparencyFallbackBorder = e.NameScope.Find<Border>("PART_TransparencyFallback");
 
             HandleTransparencyLevelChanged(PlatformImpl.TransparencyLevel);
+
+            var child = Content;
```
Comment (danwalmsley) [inline src/Avalonia.Controls/TopLevel.cs:486]: leftover?

**Review by danwalmsley — CHANGES_REQUESTED** [review]
seems to be some leftover code.

File: src/Avalonia.Controls/TopLevel.cs (hunk 2 above)
Comment (emmauss) [reply src/Avalonia.Controls/TopLevel.cs]: Yeah. Removed.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025556-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by emmauss** [comment]
Replaced `StatusBarColor` with `StatusBarTheme`

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025566-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025772-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls/Platform/ITopLevelWithPlatformStatusBar.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using Avalonia.Media;
+
+namespace Avalonia.Controls.Platform
+{
+    public interface ITopLevelWithPlatformStatusBar
```
Comment (maxkatz6) [inline src/Avalonia.Controls/Platform/ITopLevelWithPlatformStatusBar.cs:5]:
```suggestion
    [Avalonia.Metadata.Unstable]
    public interface ITopLevelWithPlatformStatusBar
```

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0025964-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request closed without merging**